File: 18_pyrm.pl
use Getopt::Std;
#use strict;
use vars '$opt_b';

getopts('b');

my $input=$ARGV[0];
my $output=">new_file.txt";
my $test=">test.txt";

if($opt_b) {open(TEST, $test);}
open(INFILE, $input);
open(OUTFILE, $output);


@pyram = ();

while (<INFILE>){
	chomp;
	push @pyram,[split /\s+/];
}
	
for $i (0..2**14-1){

	$total = 0;
	$count = 0;
	$bin=sprintf ("%b", $i);
	@turns = split(//,
	
	$bin);
	
	
	$temp = pop @turns;
	$total = 
	
}

	
	




if($opt_b) {close TEST;}
close INFILE;
close OUTFILE;

File: 18_pyrm.py
from sys import argv

script, filename = argv

file = open(filename, 'r')
line = ' '
array = []
while line:
	line = file.readline()
	line = line.rstrip()
	temp = line.split()
	temp = map(int, temp)
	if temp == []:
		break
	array.append(temp)
for x in range(len(array)-2,-1,-1):
	for y in range(len(array[x])):
		array[x][y] += max(array[x+1][y],array[x+1][y+1])
	array.pop()
	
print array